python/tests/test_wireframes.py

```diff
@@ -104,6 +106,82 @@ def callback(df: pd.DataFrame, seq_no: int):
 
         sql.run_to_completion()
 
+    def test_kafka(self):
+        from kafka import KafkaProducer
+        from kafka.admin import KafkaAdminClient, NewTopic
+        import json
+
+        KAFKA_SERVER = "localhost:19092"
+
+        print("(Re-)creating topics...")
+        admin_client = KafkaAdminClient(
+            bootstrap_servers=KAFKA_SERVER,
+            client_id="test_client"
+        )
+
+        INPUT_TOPIC = "simple_count_input"
+        OUTPUT_TOPIC = "simple_count_output"
+
+        existing_topics = set(admin_client.list_topics())
+        if INPUT_TOPIC in existing_topics:
+            admin_client.delete_topics([INPUT_TOPIC])
+        if OUTPUT_TOPIC in existing_topics:
+            admin_client.delete_topics([OUTPUT_TOPIC])
+        admin_client.create_topics([
+            NewTopic(INPUT_TOPIC, num_partitions=1, replication_factor=1),
+            NewTopic(OUTPUT_TOPIC, num_partitions=1, replication_factor=1),
+        ])
+        print("Topics ready")
+
+        # Produce of rows into the input topic
+        print("Producing rows into input topic...")
+        num_rows = 1000
+        producer = KafkaProducer(
+            bootstrap_servers=KAFKA_SERVER,
+            client_id="test_client",
+            value_serializer=lambda v: json.dumps(v).encode("utf-8"),
+        )
+        for i in range(num_rows):
+            producer.send("simple_count_input", value={"insert": {"id": i}})
+        print("Input topic contains data")
+
+        TABLE_NAME = "example"
+        VIEW_NAME = "example_count"
+
+        sql = SQLContext('kafka_test', TEST_CLIENT).get_or_create()
+        sql.register_table(TABLE_NAME, SQLSchema({"id": "INT NOT NULL PRIMARY KEY"}))
+        sql.register_view(VIEW_NAME, f"SELECT COUNT(*) as num_rows FROM {TABLE_NAME}")
+
+        PIPELINE_TO_KAFKA_SERVER = "redpanda:9092"
+
+        source_config = {
+            "topics": [INPUT_TOPIC],
+            "bootstrap.servers": PIPELINE_TO_KAFKA_SERVER,
+            "auto.offset.reset": "earliest",
+        }
+
+        kafka_format = JSONFormat().with_update_format(UpdateFormat.InsertDelete).with_array(False)
```

Comment:
It could also be a class with default values for its instantiation `JSONFormat(update_format=UpdateFormat.Raw, array=False)`, and then the user supplies arguments to change any of the defaults. Either works though.

Reply:
I am split between having a default and requiring explicit configuration. 
Having a default might mean that people just use the default and might have issues when the format doesn't match the data.

Reply:
Better to not have defaults for this, given we don't have data on what the typical format might be.